src/core/Cline.ts

```diff
@@ -1718,25 +1725,31 @@ export class Cline {
 									break
 								}
 
-								let parsedOperations: Array<{
+								type Operation = {
 									search: string
 									replace: string
 									start_line?: number
 									end_line?: number
 									use_regex?: boolean
 									ignore_case?: boolean
 									regex_flags?: string
-								}>
+								}
+								let parsedOperations: {
+									operation: Operation[] | Operation
+								}
 
 								try {
-									parsedOperations = JSON.parse(operations)
-									if (!Array.isArray(parsedOperations)) {
-										throw new Error("Operations must be an array")
+									parsedOperations = parseXml(operations, [
+										"operation.search",
+										"operation.replace",
+										"operation.regex_flags",
+									]) as {
```

Comment:
@mrubens i will pass to it like this
[
"operation.search",
"operation.replace",
"operation.regex_flags",
]